packages/docusaurus/src/webpack/utils.ts

```diff
@@ -28,38 +29,52 @@ import {
   ConfigureWebpackFn,
   ConfigurePostCssFn,
   PostCssOptions,
+  ConfigureWebpackUtils,
 } from '@docusaurus/types';
-import CssNanoPreset from '@docusaurus/cssnano-preset';
-import {version as cacheLoaderVersion} from 'cache-loader/package.json';
 import {
   BABEL_CONFIG_FILE_NAME,
   OUTPUT_STATIC_ASSETS_DIR_NAME,
 } from '../constants';
+import cssnano from 'cssnano';
+import CssNanoPreset from '@docusaurus/cssnano-preset';
+import {memoize} from 'lodash';
 
 // Utility method to get style loaders
 export function getStyleLoaders(
   isServer: boolean,
   cssOptions: {
     [key: string]: unknown;
   } = {},
-): Loader[] {
+): RuleSetRule[] {
   if (isServer) {
-    return [
-      cssOptions.modules
-        ? {
+    return cssOptions.modules
+      ? [
+          {
             loader: require.resolve('css-loader'),
             options: cssOptions,
-          }
-        : require.resolve('null-loader'),
-    ];
+          },
+        ]
+      : [
+          {
+            loader: MiniCssExtractPlugin.loader,
+            options: {
+              // Don't emit CSS files for SSR (previously used null-loader)
+              // See https://github.com/webpack-contrib/mini-css-extract-plugin/issues/90#issuecomment-811991738
+              emit: false,
+            },
+          },
+          {
+            loader: require.resolve('css-loader'),
+            options: cssOptions,
+          },
+        ];
   }
 
-  const isProd = process.env.NODE_ENV === 'production';
-  const loaders = [
+  return [
     {
       loader: MiniCssExtractPlugin.loader,
       options: {
-        hmr: !isProd,
+        esModule: false,
```

Comment:
@RDIL do you remember why you changed this?

Reply:
It broke algolia search to gen es modules (no idea exactly why, only that it works perfectly fine with this), and `hmr` is enabled by default in the new version.

Reply:
hmr is not enabled by default, you don't need to set it, because `webpack` automatically handle HMR code when you have `HotModuleReplacement` plugin

Reply:
Thanks.

I think Algolia got broken due to our v1/v2 domain switch and it's likely unrelated to this property, so going to revert it

Reply:
This broke it back in January...

Reply:
probably a temp problem because it works now

Reply:
I have no idea then, if it works I'm not going to question it 🤷‍♂️